### ISSUE TITLE
OAuth2 token request contains invalid payload

### ISSUE THREAD

**Issue by mrtaikandi** [scalar/scalar#4287]
### What happens?

When using OAuth2 with PKCE and the `authorization_code` grant type, the call to the token URL includes scope values. This results in some identity servers, such as openiddict to fail with "The 'scope' parameter is not valid in this context." error.

https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.3

```
POST https://localhost:7011
client_id=myclient&scope=openid+profile+roles&redirect_uri=https%3A%2F%2Flocalhost%3A7012%2Fopenapi%2Fv1&code=ZB4FMllF2peV9Jk6YASfAPvmwuOEidxTNNJlA&grant_type=authorization_code&code_verifier=T5Y0ZGMyegszoVxrXkzkFWRegNQQR75BfzfRe8Js
```

```
  "components": {
    "schemas": { ... },
    "securitySchemes": {
      "OAuth2": {
        "type": "oauth2",
        "flows": {
          "authorizationCode": {
            "authorizationUrl": "https://localhost:7011/connect/authorize",
            "tokenUrl": "https://localhost:7011/connect/token",
            "refreshUrl": "https://localhost:7011/connect/token",
            "scopes": {
              "openid": "OpenID",
              "profile": "Profile",
              "roles": "Roles"
            }
          }
        }
      }
    }
  }
  ```

### What did you expect to happen?

The token request not to contain the scopes.

**Comment by arielmoraes** [comment]
Do we have an ETA on this? Are PRs being accepted?

**Comment by amritk** [comment]
hello @arielmoraes PR's are being accepted. Or even just a suggestion on how we should handle this would help!

**Comment by amritk** [comment]
is it only during PKCE we don't want to send scopes? Or on all Authorization code calls

**Comment by amritk** [comment]
Does this work?

https://github.com/scalar/scalar/pull/4720

**Comment by arielmoraes** [comment]
Yes, it does partially when considering the spec, because it doesn't have the scope parameter described in the Token Endpoint section. Kevin the creator of OpenIddict explains in this comment: https://stackoverflow.com/a/78200920/2704444. 

The spec states:

> 4.1.3.  Access Token Request
> 
>    The client makes a request to the token endpoint by sending the
>    following parameters using the "application/x-www-form-urlencoded"
>    format per Appendix B with a character encoding of UTF-8 in the HTTP
>    request entity-body:
> 
>    grant_type
>          REQUIRED.  Value MUST be set to "authorization_code".
> 
>    code
>          REQUIRED.  The authorization code received from the
>          authorization server.
> 
>    redirect_uri
>          REQUIRED, if the "redirect_uri" parameter was included in the
>          authorization request as described in Section 4.1.1, and their
>          values MUST be identical.
> 
>    client_id
>          REQUIRED, if the client is not authenticating with the
>          authorization server as described in Section 3.2.1.

I used the solution he provided to solve my problem. Some security libs and frameworks tend to be very strict regarding contracts.

**Comment by amritk** [comment]
Interesting, I had read somewhere that it was optional. I think on the oauth2 playground
